packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/physics.dart';
+import 'package:vector_math/vector_math_64.dart' show Quad, Vector3, Matrix4;
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'layout_builder.dart';
+import 'scroll_physics.dart';
+import 'single_child_scroll_view.dart';
+import 'ticker_provider.dart';
+
+/// A widget that enables pan and zoom interactions with its child.
+///
+/// The user can transform the child by dragging to pan or pinching to zoom.
+///
+/// The [child] must not be null.
+///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+/// This example shows a simple Container that can be panned and zoomed.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Center(
+///     child: InteractiveViewer(
+///       boundaryMargin: EdgeInsets.all(20.0),
+///       minScale: 0.1,
+///       maxScale: 1.6,
+///       child: Container(
+///         decoration: BoxDecoration(
+///           gradient: LinearGradient(
+///             begin: Alignment.topCenter,
+///             end: Alignment.bottomCenter,
+///             colors: <Color>[Colors.orange, Colors.red],
+///             stops: <double>[0.0, 1.0],
+///           ),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+@immutable
+class InteractiveViewer extends StatefulWidget {
+  /// Create an InteractiveViewer.
+  ///
+  /// The [child] parameter must not be null.
+  InteractiveViewer({
+    Key key,
+    @required this.child,
+    this.boundaryMargin = EdgeInsets.zero,
+    this.constrained = true,
+    // These default scale values were eyeballed as reasonable limits for common
+    // use cases.
+    this.maxScale = 2.5,
+    this.minScale = 0.8,
+    this.onInteractionEnd,
+    this.onInteractionStart,
+    this.onInteractionUpdate,
+    this.scaleEnabled = true,
+    this.transformationController,
+    this.translationEnabled = true,
```

Comment:
nit: can we group the *Enabled parameters (e.g. scaleEnabled and translationEnabled)?